packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/endpoint_client.dart

```diff
@@ -0,0 +1,154 @@
+import '../../../sdk/pinpoint.dart';
+import '../shared_prefs.dart';
+import 'endpoint_global_fields_manager.dart';
+
+class EndpointClient {
+  // string userAgent = PinpointManager.class.getName + / + versionUtils.getVersion
+
+  // max event operations = 1000
+
+  // Current Endpoint being managed
+  late PublicEndpointBuilder _endpointBuilder;
+  late final EndpointGlobalFieldsManager _globalFieldsManager;
+
+  late final String _appId;
+  late final PinpointClient _pinpointClient;
+  late final SharedPrefs _sharedPrefs;
+
+  EndpointClient._getInstance(
+      this._appId,
+      this._sharedPrefs,
+      this._pinpointClient,
+      DeviceContextInfoProvider? deviceInfoProvider,
+      this._globalFieldsManager) {
+    _endpointBuilder = PublicEndpointBuilder();
+
+    _endpointBuilder.effectiveDate = DateTime.now().toIso8601String();
+    _endpointBuilder.demographic = EndpointDemographicBuilder()
+      ..appVersion = deviceInfoProvider?.appVersion
+      ..locale = deviceInfoProvider?.locale
+      ..make = deviceInfoProvider?.make
+      ..model = deviceInfoProvider?.model
+      ..modelVersion = deviceInfoProvider?.modelVersion
+      ..platform = deviceInfoProvider?.platform
+      ..platformVersion = deviceInfoProvider?.platformVersion
+      ..timezone = deviceInfoProvider?.timezone;
+    _endpointBuilder.location = EndpointLocationBuilder()
+      ..country = deviceInfoProvider?.countryCode;
+    _endpointBuilder.requestId = _sharedPrefs.getUniqueId();
+  }
+
+  // External dependencies
+  static Future<EndpointClient> getInstance(
+      String appId,
+      SharedPrefs sharedPrefs,
+      PinpointClient pinpointClient,
+      DeviceContextInfoProvider? deviceInfoProvider) async {
+    var globalFieldsManager =
+        await EndpointGlobalFieldsManager.getInstance(sharedPrefs);
+    return EndpointClient._getInstance(appId, sharedPrefs, pinpointClient,
+        deviceInfoProvider, globalFieldsManager);
+  }
+
+  void addAttribute(String name, List<String> values) =>
+      _globalFieldsManager.addAttribute(name, values);
+  void removeAttribute(String name) =>
+      _globalFieldsManager.removeAttribute(name);
+  void addMetric(String name, double value) =>
+      _globalFieldsManager.addMetric(name, value);
+  void removeMetric(String name) => _globalFieldsManager.removeMetric(name);
+
+  // update the UserProfile on the EndpointProfile
+  Future<void> setUser(
+      String userId, AnalyticsUserProfile copyFromProfile) async {
+    EndpointUserBuilder newUserBuilder = EndpointUserBuilder();
+
+    newUserBuilder.userId = userId;
+
+    if (copyFromProfile.name != null) {
+      addAttribute('name', [copyFromProfile.name!]);
+    }
+    if (copyFromProfile.email != null) {
+      addAttribute('email', [copyFromProfile.email!]);
+    }
+    if (copyFromProfile.plan != null) {
+      addAttribute('plan', [copyFromProfile.plan!]);
+    }
+
+    if (copyFromProfile.location != null) {
+      var newLoc = copyFromProfile.location!;
```

Comment:
Please update this throughout the PR.
```suggestion
      final newLoc = copyFromProfile.location!;
```